src/core/Akka/Actor/UntypedActor.cs

```diff
@@ -13,6 +18,31 @@ protected sealed override bool Receive(object message)
             return true;
         }
 
+        protected void RunTask(Action action)
+        {
+            RunTask(() =>
+            {
+                action();
+                return Task.FromResult(0);
+            });
+        }
+        protected void RunTask(Func<Task> action)
+        {
+            ActorTaskScheduler.SetCurrentState(Self, Context.Sender, null);
+
+            Task.Factory.StartNew(async () => 
+            {
+                await action().ContinueWith(x =>
+                {
+                    if (x.IsFaulted)
+                    {
+                        ExceptionDispatchInfo.Capture(x.Exception.InnerException).Throw();
+                    }
+                });
+            }, CancellationToken.None, TaskCreationOptions.None, ActorTaskScheduler.Instance);
+        }
+
```

Comment:
Thinking about it, there is a way to implement this that wouldn't be so bad:

```
protected void RunTask(Func<Task> action)
{
	var task = action();

	if (task == null)
		return;

	if (task.IsFaulted)
		ExceptionDispatchInfo.Capture(task.Exception.InnerException).Throw();

	if (task.IsCompleted)
		return;

	ActorTaskScheduler.SetCurrentState(Self, Context.Sender, null);
	Task.Factory.StartNew(async () => { await task; }, CancellationToken.None, TaskCreationOptions.None, ActorTaskScheduler.Instance);
}
```

This is just a little bit worse in performance than pure async/await, but not by much. And should retain the same principles.